using_restricted_array.rb

```diff
@@ -7,41 +7,104 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  length = 0
+  i = 0
```

Comment:
As a further optimization, you could just return `i` or index using `length` and get rid of the additional variable (either `i` or `length`).